netket/graph/symmetry.py

```diff
@@ -80,6 +81,47 @@ def remove_duplicates(self, *, return_inverse=False):
         else:
             return group
 
+    def inverse(self):
+        """
+        Returns reordered symmgroup where the each element is the inverse of
+        the original symmetry element
+
+        """
+
+        automorphisms = self.to_array()
+        n_symm = len(automorphisms)
+        inverse = np.zeros([n_symm], dtype=int)
+        automorphisms = np.array(automorphisms)
+        for i, perm1 in enumerate(automorphisms):
+            for j, perm2 in enumerate(automorphisms):
+                perm_sq = perm1[perm2]
+                if np.all(perm_sq == np.arange(len(perm_sq))):
+                    inverse[i] = j
+
+        return SymmGroup([self.elems[i] for i in inverse], self.graph)
+
+    def group_algebra(self):
+        """
+        Computes an array relative displacements between poses in automorphisms, s.t.
+        group_algebra[i,j] = inverse[i]*automorphisms[j]
+
+        This array is flattened and converted to a tuple before returning
+
+        """
+
+        automorphisms = self.to_array()
+        n_symm = len(automorphisms)
+        inverse = self.inverse().to_array()
+        group_algebra = np.zeros([n_symm, n_symm], dtype=int)
+
+        for i, inv in enumerate(inverse):
```

Comment:
oh I see, it's this triple loop + an if running an all the vector!!... uhm isn't there a way to rewrite this is a faster way?

Reply:
I'm sure there's a solution with hashing that is a lot faster. I'm trying to identify the indices of duplicates between a list of length of N and a list of length N^2 (which I can make with arrays). I think I can make a dict with the N^2 list and loop over the N list?

Reply:
Yeah, I think I have to hash over the length N list but this should still be fast enough

Reply:
Okay, problem has been solved, it's a lot faster than graph.automorphisms(), and scales approximately like 17microseconds*n^2. So even for a 10x10 triangular that is like 25 seconds

Reply:
Still, could you check if putting it in a numba jit block makes it even a bit faster?